src/Locale/index.js

```diff
@@ -1,116 +1,38 @@
- *    > Locale.load('static/locale/locale.json');
- *    > Locale.setLanguage('en');
- *
- * 3. Use localization strings:
- *    > console.log(Locale.tr.how);
- *    How do you want your egg today?
- *    > console.log(Locale.tr.boiledEgg);
- *    Boiled egg
- *
- * 4. String formatting
- *    > console.log(Locale.tr.buyQuestion.format(10, 0.5));
- *    I'd like to buy 10 eggs, 0.5 dollars each.
- */
-
-class Locale {
-  constructor() {
-    this.__enabled = false
-  }
-
-  /**
-   * Loads translation object from external json file.
-   *
-   * @param {String} path Path to resource.
-   * @return {Promise}
-   */
-  async load(path) {
-    if (!this.__enabled) {
-      return
-    }
-
-    await fetch(path)
-      .then(resp => resp.json())
-      .then(resp => {
-        this.loadFromObject(resp)
-      })
-  }
-
-  /**
-   * Sets language used by module.
-   *
-   * @param {String} lang
-   */
-  setLanguage(lang) {
-    this.__enabled = true
-    this.language = lang
-  }
-
-  /**
-   * Returns reference to translation object for current language.
-   *
-   * @return {Object}
-   */
-  get tr() {
-    return this.__trObj[this.language]
-  }
-
-  /**
-   * Loads translation object from existing object (binds existing object).
-   *
-   * @param {Object} trObj
-   */
-  loadFromObject(trObj) {
-    this.__trObj = trObj
-    for (const lang of Object.values(this.__trObj)) {
-      for (const str of Object.keys(lang)) {
-        lang[str] = new LocalizedString(lang[str])
-      }
-    }
-  }
+import Locale from './'
+import Profile from '../Profile/index'
+import { loadTranslationFile, format, isCorrectLocale } from './helpers'
+
+const defaultLocale = 'en-US'
+let loadedLanguageFile = undefined
+
+// Init
+export const initLocale = async () => {
+  return Profile.locale().then(locale => {
+    return Locale.setLocale(locale).then(isSet => {
+      return isSet
+    })
+  })
 }
 
-/**
- * Extended string class used for localization.
- */
-class LocalizedString extends String {
-  /**
-   * Returns formatted LocalizedString.
-   * Replaces each placeholder value (e.g. {0}, {1}) with corresponding argument.
-   *
-   * E.g.:
-   * > new LocalizedString('{0} and {1} and {0}').format('A', 'B');
-   * A and B and A
-   *
-   * @param  {...any} args List of arguments for placeholders.
-   */
-  format(...args) {
-    const sub = args.reduce((string, arg, index) => string.split(`{${index}}`).join(arg), this)
-    return new LocalizedString(sub)
-  }
+// public API Locale
+export default {
+  get(attributeName, ...args) {
+    if (!(loadedLanguageFile && loadedLanguageFile[attributeName])) return attributeName
+    if (args && args.length > 0) return format(loadedLanguageFile[attributeName], args)
+    return loadedLanguageFile[attributeName]
+  },
+
+  // Set ISO locale code aka: 'en-US, nl-NL, etc' default is en-US.
+  setLocale(locale) {
+    return loadTranslationFile(isCorrectLocale(locale, defaultLocale), defaultLocale).then(
```

Comment:
I would move the call to `isCorrectLocale` to the `loadTranslationFile` method itself (then you can also remove the import of isCorrectLocale)

Reply:
Moved to `loadTranslationFile`